### PR TITLE
fix(svdconv): recognize CA55 and correct Cortex-A IRQ limits

### PR DESCRIPTION
**Pull request by thorstendb-ARM**
## Fixes

- Addresses the missing `CA55` CPU-name entry and incorrect Cortex-A IRQ limit reported in #2201.

**Cortex-A SVDs are not officially supported by SVDConv. This PR corrects the existing CPU list and interrupt-limit metadata; it does not introduce official Cortex-A device or generator support.**

## Changes

- Add `CA55` to the shared enum/parser table and CPU feature table.
- Correct the exclusive interrupt-number limits for the known integrated GICs:
  - CA5, CA9 and CA15: 256.
  - CA7: 512.
  - Document the scope and Arm manual references alongside the table.
- Define `0` as "no CPU-based IRQ limit is known" for CA8, CA17, CA53, CA55, CA57, CA72 and OTHER.
- Skip only the CPU-limit checks (`M331` and `M389`) when the limit is unknown.
- Preserve explicit `deviceNumInterrupts` validation (`M381`), missing IRQ-value errors (`M330`), and existing missing/unknown CPU diagnostics and fallbacks.
- Add 61 regression test cases covering fixed boundaries, unknown limits, explicit device limits, missing values, CPU fallbacks and CA55 names.

Existing Cortex-M-oriented validation, including `M355`, and the generators remain unchanged. No schema changes, third-party dependency changes, or repository-wide build/workflow changes are included.

## Validation

- macOS arm64 Release build: successful.
- Final tree: 69 unit tests and 21 integration tests pass; two pre-existing disabled unit tests remain disabled.
- Targeted CA55/IRQ249 validation and header-output probe: accepted without CPU-name or CPU-limit errors. An explicit device limit of 249 still rejects IRQ249; a limit of 250 accepts it.
- Temporary IRQ-limit regression audit: 353 installed pack SVDs, baseline/current in normal and strict modes (1,412 runs), with identical diagnostics and exit codes and no crashes or timeouts. This audit preceded the final CA55 entry; the installed corpus contains no Cortex-A/OTHER SVDs.
- Additional temporary synthetic tests using real peripheral data: 13 CPU/limit variants, 52 runs, all expected `M331`/`M389`/`M381`/`M330` counts verified.
- The temporary audit scripts, modified test copies and captured logs are not included in the PR; original pack files were not modified.

## Checklist

- [x] 🤖 This change is covered by unit tests (if applicable).
- [x] 🤹 Manual testing has been performed (if necessary).
- [x] 🛡️ Security impacts have been considered (if relevant).
- [x] 📖 Documentation updates are complete (if required): limit semantics and source references are documented in the code.
- [x] 🧠 Third-party dependencies and TPIP updated (if required): no changes required.

### CODE REVIEW

File: tools/svdconv/test/unittests/src/SvdInterruptTest.cpp [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+INSTANTIATE_TEST_SUITE_P(CpuInterruptLimits, SvdInterruptTest,
+  testing::Values(
+    InterruptLimit{"CA5", SvdTypes::CpuType::CA5, 256U},
+    InterruptLimit{"CA7", SvdTypes::CpuType::CA7, 512U},
+    InterruptLimit{"CA9", SvdTypes::CpuType::CA9, 256U},
+    InterruptLimit{"CA15", SvdTypes::CpuType::CA15, 256U},
+    InterruptLimit{"CM3", SvdTypes::CpuType::CM3, 240U},
+    InterruptLimit{"CM33", SvdTypes::CpuType::CM33, 480U}),
+  [](const testing::TestParamInfo<InterruptLimit>& info) { return info.param.name; });
```
[the remark this answers is not in the document]
Comment (thorstendb-ARM) [reply tools/svdconv/test/unittests/src/SvdInterruptTest.cpp:142]: This is a false positive in the GoogleTest `INSTANTIATE_TEST_SUITE_P` expansion, not an unused SVDConv function.

The macro [passes both helper function addresses to `AddTestSuiteInstantiation()`](https://github.com/google/googletest/blob/6910c9d9165801d8827d628cb72eb7ea9dd538c5/googletest/include/gtest/gtest-param-test.h#L524-L528). GoogleTest stores these callbacks and [invokes them from `RegisterTests()`](https://github.com/google/googletest/blob/6910c9d9165801d8827d628cb72eb7ea9dd538c5/googletest/include/gtest/internal/gtest-param-util.h#L554-L581) to generate parameter values and test names.

Here, `CpuInterruptLimits/SvdInterruptTest` instantiates 30 cases (5 tests across 6 CPU types). These helpers are required for that registration; removing them would break the test suite. No code change is needed.

File: tools/svdconv/test/unittests/src/SvdInterruptTest.cpp [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+INSTANTIATE_TEST_SUITE_P(NoCpuInterruptLimit, SvdNoCpuInterruptLimitTest,
+  testing::Values(
+    InterruptLimit{"CA8", SvdTypes::CpuType::CA8, CpuFeature::NO_CPU_INTERRUPT_LIMIT},
+    InterruptLimit{"CA17", SvdTypes::CpuType::CA17, CpuFeature::NO_CPU_INTERRUPT_LIMIT},
+    InterruptLimit{"CA53", SvdTypes::CpuType::CA53, CpuFeature::NO_CPU_INTERRUPT_LIMIT},
+    InterruptLimit{"CA55", SvdTypes::CpuType::CA55, CpuFeature::NO_CPU_INTERRUPT_LIMIT},
+    InterruptLimit{"CA57", SvdTypes::CpuType::CA57, CpuFeature::NO_CPU_INTERRUPT_LIMIT},
+    InterruptLimit{"CA72", SvdTypes::CpuType::CA72, CpuFeature::NO_CPU_INTERRUPT_LIMIT},
+    InterruptLimit{"OTHER", SvdTypes::CpuType::OTHER, CpuFeature::NO_CPU_INTERRUPT_LIMIT}),
+  [](const testing::TestParamInfo<InterruptLimit>& info) { return info.param.name; });
```
[the remark this answers is not in the document]
Comment (thorstendb-ARM) [reply tools/svdconv/test/unittests/src/SvdInterruptTest.cpp:187]: This GoogleTest-generated `dummy_` variable intentionally has an unread value. Its namespace-scope initializer performs the required parameterized-test registration.

GoogleTest already marks it with [`GTEST_INTERNAL_ATTRIBUTE_MAYBE_UNUSED`](https://github.com/google/googletest/blob/6910c9d9165801d8827d628cb72eb7ea9dd538c5/googletest/include/gtest/gtest-param-test.h#L517-L528). [`AddTestSuiteInstantiation()`](https://github.com/google/googletest/blob/6910c9d9165801d8827d628cb72eb7ea9dd538c5/googletest/include/gtest/internal/gtest-param-util.h#L541-L547) returns 0 solely to support this initialization; the registration side effect is what matters.

For `NoCpuInterruptLimit/SvdNoCpuInterruptLimitTest`, that registration provides 28 cases (4 tests across 7 CPU types). Deleting the declaration would remove the registration, not merely an unused integer. This is a framework-related false positive; no code change is needed.

File: tools/svdconv/test/unittests/src/SvdInterruptTest.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (thorstendb-ARM) [reply tools/svdconv/test/unittests/src/SvdInterruptTest.cpp:142]: This GoogleTest-generated `dummy_` variable intentionally has an unread value. Its namespace-scope initializer performs the required parameterized-test registration.

GoogleTest already marks it with [`GTEST_INTERNAL_ATTRIBUTE_MAYBE_UNUSED`](https://github.com/google/googletest/blob/6910c9d9165801d8827d628cb72eb7ea9dd538c5/googletest/include/gtest/gtest-param-test.h#L517-L528). [`AddTestSuiteInstantiation()`](https://github.com/google/googletest/blob/6910c9d9165801d8827d628cb72eb7ea9dd538c5/googletest/include/gtest/internal/gtest-param-util.h#L541-L547) returns 0 solely to support this initialization; the registration side effect is what matters.

For `CpuInterruptLimits/SvdInterruptTest`, that registration provides 30 cases (5 tests across 6 CPU types). Deleting the declaration would remove the registration, not merely an unused integer. This is a framework-related false positive; no code change is needed.

File: tools/svdconv/test/unittests/src/SvdInterruptTest.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (thorstendb-ARM) [reply tools/svdconv/test/unittests/src/SvdInterruptTest.cpp:187]: This is a false positive in the GoogleTest `INSTANTIATE_TEST_SUITE_P` expansion, not an unused SVDConv function.

The macro [passes both helper function addresses to `AddTestSuiteInstantiation()`](https://github.com/google/googletest/blob/6910c9d9165801d8827d628cb72eb7ea9dd538c5/googletest/include/gtest/gtest-param-test.h#L524-L528). GoogleTest stores these callbacks and [invokes them from `RegisterTests()`](https://github.com/google/googletest/blob/6910c9d9165801d8827d628cb72eb7ea9dd538c5/googletest/include/gtest/internal/gtest-param-util.h#L554-L581) to generate parameter values and test names.

Here, `NoCpuInterruptLimit/SvdNoCpuInterruptLimitTest` instantiates 28 cases (4 tests across 7 CPU types). These helpers are required for that registration; removing them would break the test suite. No code change is needed.

**Review by jkrech — APPROVED** [review]
LGTM
